handlers/mesh.go

```diff
@@ -63,3 +67,13 @@ func MeshGraph(
 		respond(w, code, payload)
 	}
 }
+
+func filterAccessibleControlPlanes(ctx context.Context, namespaceService business.NamespaceService, mesh *models.Mesh) {
+	for i, cp := range mesh.ControlPlanes {
+		// Check if the user is able to access to the control plane
+		_, err := namespaceService.GetClusterNamespace(ctx, cp.IstiodNamespace, cp.Cluster.Name)
+		if err != nil && strings.Contains(err.Error(), "forbidden") {
+			mesh.ControlPlanes = append(mesh.ControlPlanes[:i], mesh.ControlPlanes[i+1:]...)
```

Comment:
I think there's a nifty func for this now if you want to use it: https://pkg.go.dev/slices#Delete